iml-gui/crate/src/components/command_modal.rs

```diff
@@ -64,87 +115,125 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
+                    let temp_slice = cmds.iter().map(|x: &Arc<Command>| (**x).clone()).collect::<Vec<_>>();
+                    model.assign_commands(&temp_slice);
+                    if !is_all_finished(&model.commands) {
+                        orders.send_msg(Msg::FetchTree);
                     }
                 }
                 Input::Ids(ids) => {
-                    model.loading = true;
-
-                    orders.perform_cmd(fetch_command_status(ids));
+                    // we have ids only, so we need to populate the vector first
+                    orders.perform_cmd(fetch_the_batch(ids, |x| Msg::FetchedCommands(Box::new(x))));
                 }
             }
         }
-        Msg::Fetch => {
-            model.cancel = None;
-
-            if !model.commands.is_empty() {
-                let ids = model.commands_all.iter().map(|x| x.id).collect();
-                orders.skip().perform_cmd(fetch_command_status(ids));
+        Msg::FetchTree => {
+            model.tree_cancel = None;
+            if !is_all_finished(&model.commands) {
+                schedule_fetch_tree(model, orders);
             }
         }
-        Msg::Fetched(cmd_status_result) => {
-            model.loading = false;
-
-            match *cmd_status_result {
-                Ok(cmd_status) => {
-                    model.commands_all = cmd_status.objects.into_iter().map(Arc::new).collect();
-                    model.commands = model
-                        .commands_all
-                        .clone()
-                        .into_iter()
-                        .filter(|x| !is_finished(x))
-                        .collect();
+        Msg::FetchedCommands(commands_data_result) => {
+            match *commands_data_result {
+                Ok(api_list) => {
+                    model.assign_commands(&api_list.objects);
                 }
                 Err(e) => {
-                    error!("Failed to perform fetch_command_status {:#?}", e);
-
+                    error!("Failed to perform FetchedCommands {:#?}", e);
                     orders.skip();
                 }
             }
-            if !model.commands.is_empty() {
-                let (cancel, fut) = sleep_with_handle(POLL_INTERVAL, Msg::Fetch, Msg::Noop);
-
-                model.cancel = Some(cancel);
+            if !is_all_finished(&model.commands) {
+                let (cancel, fut) = sleep_with_handle(POLL_INTERVAL, Msg::FetchTree, Msg::Noop);
+                model.tree_cancel = Some(cancel);
                 orders.perform_cmd(fut);
             }
         }
-        Msg::Open(x) => {
-            model.opens.insert(x);
-        }
-        Msg::Close(x) => {
-            model.opens.remove(&x);
+        Msg::FetchedJobs(jobs_data_result) => match *jobs_data_result {
+            Ok(api_list) => {
+                model.assign_jobs(&api_list.objects);
+            }
+            Err(e) => {
+                error!("Failed to perform FetchJobs {:#?}", e);
```

Comment:
```suggestion
                error!(format!("Failed to perform FetchJobs {:#?}", e));
```
Otherwise it will print `"Failed to perform FetchJobs {:#?}", ...`

Reply:
Thanks!